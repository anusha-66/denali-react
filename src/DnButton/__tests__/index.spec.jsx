/**
 * Copyright 2020, Verizon Media
 * Licensed under the terms of the MIT license. See accompanying LICENSE.md file for terms.
 */

import React from 'react';
import { mount, render } from 'enzyme';
import { DnButton, DnButtonStyles, DnButtonSizes, DnButtonTypes } from '../../index'; // src/index.tsx

describe('DnButton Tests', () => {
    it('should export DnButtonStyles', () => {
        expect.assertions(1);

        expect(DnButtonStyles).toStrictEqual({
            ghost: 'ghost',
            outline: 'outline',
            solid: 'solid',
            text: 'text',
        });
    });

    it('should export DnButtonSizes', () => {
        expect.assertions(1);

        expect(DnButtonSizes).toStrictEqual({
            default: 'default',
            large: 'large',
            medium: 'medium',
            small: 'small',
        });
    });

    it('should export DnButtonTypes', () => {
        expect.assertions(1);

        expect(DnButtonTypes).toStrictEqual({
            button: 'button',
            reset: 'reset',
            submit: 'submit',
        });
    });

    it('should render with default props', () => {
        expect.assertions(1);

        const rendered = render(<DnButton>Continue</DnButton>);

        expect(rendered).toMatchSnapshot();
    });

    it('should render correctly with custom props', () => {
        expect.assertions(1);

        const rendered = render(
            <DnButton
                isActive
                style={DnButtonStyles.ghost}
                size={DnButtonSizes.large}
                isInverse
                isDisabled
                isFull
                type={DnButtonTypes.submit}
                className="p-18"
            >
                Submit From
            </DnButton>,
        );

        expect(rendered).toMatchSnapshot();
    });

    it('should render correctly with custom props with is loading', () => {
        expect.assertions(1);

        const rendered = render(
            <DnButton
                isActive
                style={DnButtonStyles.ghost}
                size={DnButtonSizes.large}
                isInverse
                isDisabled
                isFull
                type={DnButtonTypes.submit}
                isLoading
                className="p-18"
            >
                Submit From
            </DnButton>,
        );

        expect(rendered).toMatchSnapshot();
    });

    it('should call on click correctly', () => {
        expect.assertions(1);

        const mockedOnClick = jest.fn();
        const wrapper = mount(<DnButton onClick={mockedOnClick}>Button</DnButton>);
        wrapper.find('button').simulate('click');

        expect(mockedOnClick).toHaveBeenCalledTimes(1);
    });

    it('should call on focus correctly', () => {
        expect.assertions(1);

        const mockedOnFocus = jest.fn();
        const wrapper = mount(<DnButton onFocus={mockedOnFocus}>Button</DnButton>);
        wrapper.find('button').simulate('focus');

        expect(mockedOnFocus).toHaveBeenCalledTimes(1);
    });

    it('should call on keydown correctly', () => {
        expect.assertions(1);

        const mockedOnKeydown = jest.fn();
        const wrapper = mount(<DnButton onKeyDown={mockedOnKeydown}>Button</DnButton>);
        wrapper.find('button').simulate('keydown');

        expect(mockedOnKeydown).toHaveBeenCalledTimes(1);
    });
});
