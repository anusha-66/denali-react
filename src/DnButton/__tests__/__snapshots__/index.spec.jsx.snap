// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DnButton Tests should render correctly with custom props 1`] = `
<button
  class="button is-active is-ghost is-large is-inverse is-disabled is-full p-18"
  type="submit"
>
  Submit From
</button>
`;

exports[`DnButton Tests should render correctly with custom props with is loading 1`] = `
<button
  class="button is-active is-ghost is-large is-inverse is-disabled is-full p-18"
  type="submit"
>
  <div
    class="loader loader--button"
  />
</button>
`;

exports[`DnButton Tests should render with default props 1`] = `
<button
  class="button is-solid is-default"
  type="button"
>
  Continue
</button>
`;
